update API using ZAP 2.12.0

njmulsqb opened the pull request:
Signed-off-by: Najam Ul Saqib <najamulsaqib@tutamail.com>

Here's a PR that includes updated ZAP code. I am creating a PR just to add new features to the outdated NodeJS API.

I tried to update the generated files and followed steps from https://github.com/zaproxy/zap-api-python#updating-the-generated-files
Here are the steps I took:
1. Ran `./gradlew generateNodeJSApiEndpoints` in zaproxy directory
2. Added the newly added files to `index.js` following the pattern

If I am missing anything let me know.

thc202 commented:
The changelog should be updated. The add-on APIs should be regenerated as well.

njmulsqb commented:
I tried to ran `generateNodeJsZapApiClientFiles` in zap-extensions but it is failing with an error, check this https://pastebin.com/2gTE7REq
As for changelog, what section needs to be changed and what version will this one be? 1.1?

thc202 commented:
Add the flag `--continue`. For the changelog, mention the update for 2.12 and the add-ons added/updated, e.g. https://github.com/zaproxy/zap-api-python/blob/master/CHANGELOG.md (the version should stay as Unreleased, that's changed during the release).

njmulsqb commented:
I used the `--continue` with it, it showed lots of failure errors but it also added new files in the API, if you're interested here's the trace https://pastebin.com/xX50QymF

I have updated the changelog, cant really locate their versions so just mentioned the name, if version is required let me know where can I find it. 

I have created an account on NPM with username `njmulsqb` if thats required to update the NPM package. (I am amazed to see 3k+ weekly downloads on it)

njmulsqb commented:
I dont know why it is asking me to pull each time I try to push, who is making the changes on remote repo? :P

kingthorin commented:
It’s likely because you’re working on master. You should have created a new branch before starting your changes.

njmulsqb commented:
@kingthorin Yeah, realized it very late in the process. Any workaround?

kingthorin commented:
Don’t think so. You could just ignore the message.

thc202 commented:
The versions of the add-ons are in the corresponding `gradle.properties` files, e.g.: https://github.com/zaproxy/zap-extensions/blob/0d6fcd3f5f9fe90860e8ebd1d6d3b580756fd3bb/addOns/accessControl/gradle.properties Although the changelog is probably a better place to check the released version.

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased]
 
+## [1.1.0] - 2022-11-26 - Upgrade API to ZAP 2.12.0
```
This should be removed.

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+
+### Added
+* Add the API of the following Add-Ons
+    * Access Control version 9
```
`8`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+### Added
+* Add the API of the following Add-Ons
+    * Access Control version 9
+    * Alert Filter version 15
```
`Alert Filters version 14`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+* Add the API of the following Add-Ons
+    * Access Control version 9
+    * Alert Filter version 15
+    * Automation version 0.20.0
```
`Automation Framework version 0.19.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Access Control version 9
+    * Alert Filter version 15
+    * Automation version 0.20.0
+    * Exim version 0.4.0
```
`Import/Export version 0.3.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Alert Filter version 15
+    * Automation version 0.20.0
+    * Exim version 0.4.0
+    * GraphQL version 0.13.0
```
`GraphQL Support version 0.12.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Automation version 0.20.0
+    * Exim version 0.4.0
+    * GraphQL version 0.13.0
+    * Network version 0.6.0
```
`0.5.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Exim version 0.4.0
+    * GraphQL version 0.13.0
+    * Network version 0.6.0
+    * Reports version 0.18.0
```
`Report Generation version 0.17.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * GraphQL version 0.13.0
+    * Network version 0.6.0
+    * Reports version 0.18.0
+    * Retest version 0.5.0
```
`0.4.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Network version 0.6.0
+    * Reports version 0.18.0
+    * Retest version 0.5.0
+    * Revisit version 5
```
`4`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Reports version 0.18.0
+    * Retest version 0.5.0
+    * Revisit version 5
+    * Wappalyzer version 21.17.0
```
`Wappalyzer - Technology Detection version 21.16.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+### Changed
+* Update Core APIs for 2.12.0
+* Update the APIs of following Add-Ons
+    * Ajax Spider version 23.11.0
```
`23.10.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+* Update Core APIs for 2.12.0
+* Update the APIs of following Add-Ons
+    * Ajax Spider version 23.11.0
+    * OpenAPI version 31
```
`OpenAPI Support version 30`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+* Update the APIs of following Add-Ons
+    * Ajax Spider version 23.11.0
+    * OpenAPI version 31
+    * PNH version 14
```
`Plug-n-Hack Configuration version 13`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Ajax Spider version 23.11.0
+    * OpenAPI version 31
+    * PNH version 14
+    * Replacer version 12
```
`11`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * OpenAPI version 31
+    * PNH version 14
+    * Replacer version 12
+    * Reveal version 6
```
`5`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * PNH version 14
+    * Replacer version 12
+    * Reveal version 6
+    * Selenium version 15.12.0
```
`15.11.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Replacer version 12
+    * Reveal version 6
+    * Selenium version 15.12.0
+    * SOAP version 17
```
`SOAP Support version 16`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Reveal version 6
+    * Selenium version 15.12.0
+    * SOAP version 17
+    * Spider version 0.2.0
```
`0.1.0`

thc202 commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+    * Selenium version 15.12.0
+    * SOAP version 17
+    * Spider version 0.2.0
+    * Web Socket version 28
```
`WebSockets version 27`

njmulsqb commented:
Versions corrected!

njmulsqb commented:
When is it expected to merge?

thc202 commented:
I guess after the merge of the other PR, no point merging this if it's going to be rewritten.

njmulsqb commented:
Agreed, 2.12 should be only generated now once all the breaking changes are implemented, like the PR you mentioned and the HTTP library one.
